diff --git a/src/test/org/apache/commons/math/fraction/FractionTest.java b/src/test/org/apache/commons/math/fraction/FractionTest.java
index 986ffc235..5892ae22e 100644
--- a/src/test/org/apache/commons/math/fraction/FractionTest.java
+++ b/src/test/org/apache/commons/math/fraction/FractionTest.java
@@ -17,7 +17,7 @@
 package org.apache.commons.math.fraction;
 
 import org.apache.commons.math.ConvergenceException;
-
+import org.junit.Assert;
 import junit.framework.TestCase;
 
 /**
@@ -25,6 +25,15 @@ import junit.framework.TestCase;
  */
 public class FractionTest extends TestCase {
 
+    private void checkIntegerOverflow(double a) {
+        try {
+            new Fraction(a, 1.0e-12, 1000);
+            Assert.fail("an exception should have been thrown");
+        } catch (ConvergenceException ce) {
+            // expected behavior
+        }
+    }
+
     private void assertFraction(int expectedNumerator, int expectedDenominator, Fraction actual) {
         assertEquals(expectedNumerator, actual.getNumerator());
         assertEquals(expectedDenominator, actual.getDenominator());
